pkg/admin/facade/user_profile.go

```diff
@@ -49,6 +50,14 @@ func (f *UserProfileFacade) UpdateUserProfile(
 	customAttrs map[string]interface{},
 ) (err error) {
 	updated := false
+	user, err := f.User.GetRaw(userID)
+	if err != nil {
+		return err
+	}
+	if user.IsAnonymized {
+		return ErrUserIsAnonymized
+	}
+
```

Comment:
If we want to guard against mutation on anonymised user, we have to guard against all possible mutations. We can find the list in the schema. Maybe we need to introduce a facade to ensure all mutations cannot be applied to an anonymised user.